core/rts.cpp

```diff
@@ -30,6 +32,7 @@ void RelationalTransitionSystem::set_behavior(const Term & init,
   }
   init_ = init;
   trans_ = trans;
+  set_updated_states(trans_);
```

Comment:
You could also just remove this call as well as the assignment and check above and just call `set_trans`.